routers/user/home.go

```diff
@@ -307,10 +308,23 @@ func Issues(ctx *context.Context) {
 		return
 	}
 
+	var issuesStates = make([]*models.CommitStatus, 0, len(issues))
 	for _, issue := range issues {
 		issue.Repo = showReposMap[issue.RepoID]
+
+		if issue.IsPull {
+			issue.LoadAttributes()
+			statuses, err := models.GetLatestCommitStatus(issue.Repo, issue.PullRequest.MergeBase, 0)
```

Comment:
Can you extract this block to new function? I've found this (almost) same code block 3 times in existing codebase.

Reply:
I think they are different `for ... range`.

Reply:
It's same, the only difference is you are adding result of `models.CalcCommitStatus(statuses)` to slice. Instead of calling 2 functions and 1 err check, you can call just 1 function (err check can be inside too, it changes nothing). Best thing would be to rename `GetLatestCommitStatus` to `GetCommitStatuses` (as it returns array, not just 1 status) and create/use `GetLatestCommitStatus` to return just one status (for now, this few lines of code can be used, but I want to change it to pure SQL select).

Reply:
ok, `GetCommitStatuses` is here also, I don't understand `GetLatestCommitStatus` implementation... As I understand, it should return only 1 status in slice.